### ISSUE TITLE
Every youtube video broken

### ISSUE THREAD

**Issue by likeawindrammer** [TeamNewPipe/NewPipe-legacy#20]
- [x] I carefully read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md) and agree to them.
- [x] I checked if the issue/feature exists in the latest version.
- [x] I did use the [incredible bugreport to markdown converter](https://teamnewpipe.github.io/CrashReportToMarkdown/) to paste bug reports.

## Exception
* __User Action:__ requested stream
* __Request:__ https://www.youtube.com/watch?v=frvV6LnNM0g
* __Content Language:__ US
* __Service:__ YouTube
* __Version:__ 0.18.1-rc.1
* __OS:__ Linux Android 4.3 - 18

<details><summary><b>Crash log</b></summary><p>

```
org.schabi.newpipe.extractor.services.youtube.extractors.YoutubeStreamExtractor$DecryptException: Could not parse decrypt function 
	at org.schabi.newpipe.extractor.services.youtube.extractors.YoutubeStreamExtractor.loadDecryptionCode(YoutubeStreamExtractor.java:849)
	at org.schabi.newpipe.extractor.services.youtube.extractors.YoutubeStreamExtractor.onFetchPage(YoutubeStreamExtractor.java:716)
	at org.schabi.newpipe.extractor.Extractor.fetchPage(Extractor.java:54)
	at org.schabi.newpipe.extractor.stream.StreamInfo.getInfo(StreamInfo.java:64)
	at org.schabi.newpipe.extractor.stream.StreamInfo.getInfo(StreamInfo.java:60)
	at org.schabi.newpipelegacy.util.ExtractorHelper.lambda$getStreamInfo$3(ExtractorHelper.java:115)
	at org.schabi.newpipelegacy.util.-$$Lambda$ExtractorHelper$i-hFeJnSJwqTaktlkxwfnGBKECA.call(lambda)
	at io.reactivex.internal.operators.single.SingleFromCallable.subscribeActual(SingleFromCallable.java:44)
	at io.reactivex.Single.subscribe(Single.java:3438)
	at io.reactivex.internal.operators.single.SingleDoOnSuccess.subscribeActual(SingleDoOnSuccess.java:35)
	at io.reactivex.Single.subscribe(Single.java:3438)
	at io.reactivex.internal.operators.maybe.MaybeFromSingle.subscribeActual(MaybeFromSingle.java:41)
	at io.reactivex.Maybe.subscribe(Maybe.java:4154)
	at io.reactivex.internal.operators.maybe.MaybeConcatArray$ConcatMaybeObserver.drain(MaybeConcatArray.java:153)
	at io.reactivex.internal.operators.maybe.MaybeConcatArray$ConcatMaybeObserver.request(MaybeConcatArray.java:78)
	at io.reactivex.internal.operators.flowable.FlowableElementAtMaybe$ElementAtSubscriber.onSubscribe(FlowableElementAtMaybe.java:66)
	at io.reactivex.internal.operators.maybe.MaybeConcatArray.subscribeActual(MaybeConcatArray.java:42)
	at io.reactivex.Flowable.subscribe(Flowable.java:14479)
	at io.reactivex.internal.operators.flowable.FlowableElementAtMaybe.subscribeActual(FlowableElementAtMaybe.java:36)
	at io.reactivex.Maybe.subscribe(Maybe.java:4154)
	at io.reactivex.internal.operators.maybe.MaybeToSingle.subscribeActual(MaybeToSingle.java:46)
	at io.reactivex.Single.subscribe(Single.java:3438)
	at io.reactivex.internal.operators.single.SingleSubscribeOn$SubscribeOnObserver.run(SingleSubscribeOn.java:89)
	at io.reactivex.Scheduler$DisposeTask.run(Scheduler.java:578)
	at io.reactivex.internal.schedulers.ScheduledRunnable.run(ScheduledRunnable.java:66)
	at io.reactivex.internal.schedulers.ScheduledRunnable.call(ScheduledRunnable.java:57)
	at java.util.concurrent.FutureTask.run(FutureTask.java:234)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:153)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:267)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1080)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:573)
	at java.lang.Thread.run(Thread.java:841)
Caused by: org.schabi.newpipe.extractor.utils.Parser$RegexException: failed to find pattern ";([A-Za-z0-9_\$]{2})\...\( inside of var lea=function(a){a=a.split("");var b=[function(c,d){d=(d%c.length+c.length)%c.length;c.splice(-d).reverse().forEach(function(e){return c.unshift(e)};"
	at org.schabi.newpipe.extractor.utils.Parser.matchGroup(Parser.java:73)
	at org.schabi.newpipe.extractor.utils.Parser.matchGroup(Parser.java:61)
	at org.schabi.newpipe.extractor.utils.Parser.matchGroup1(Parser.java:52)
	at org.schabi.newpipe.extractor.services.youtube.extractors.YoutubeStreamExtractor.loadDecryptionCode(YoutubeStreamExtractor.java:836)
	... 31 more

```
</p></details>
<hr>

**Comment by yura04** [comment]
Same as this https://github.com/TeamNewPipe/NewPipe/issues/2981
In the non-legacy version there's an update. Hope author will add it here soon

**Comment by john3voltas** [comment]
Yep, same issue here. SkyTube also having issues.
I hope the devs can fix it soon.

**Comment by theScrabi** [comment]
Its fixed already. NewPipe 0.18.2 could fix it. SkyTube should also update to 0.18.2 extractor.

**Comment by likeawindrammer** [comment]
@theScrabi Yes, it is fixed on NewPipe 0.18.2. But NewPipe *legacy* still need a rebuild to get the fix.

**Comment by likeawindrammer** [comment]
Fixed indeed with v0.18.2. Thank you very much @friendlyanon for maintaining this version for the old yet well built android devices, thank you @blackbox87 for the ui fixes and thank you to everyone else in the newpipe dev team.

**Comment by blackbox87** [comment]
You're welcome, although I noticed that @friendlyanon thanked me for "fixes for 4.0 and 4.1 devices", but my fixes actually apply to a much wider range of devices. And NewPipe itself doesn't actually support Android 4.0 as the minimum API level is 16.

- InstantiationException was an error you'd receive on all devices when going into Settings > Content > PeerTube instances.
- NoClassDefFoundError was an error you'd receive on all devices older than Android 4.4 when backing out of a video.
- Radio buttons weren't styled correctly on all devices older than Lollipop after migrating to AndroidX.
- I enabled TLS v1.1/1.2 for devices stuck on Jelly Bean.
- Position indicators weren't displayed correctly on devices running Android 4.1.

**Comment by blackbox87** [comment]
You're welcome, although I noticed that @friendlyanon thanked me for "fixes for 4.0 and 4.1 devices", but my fixes actually apply to a much wider range of devices. And NewPipe itself doesn't actually support Android 4.0 as the minimum API level is 16.

- InstantiationException was an error you'd receive on all devices when going trying to configure PeerTube instances.
- NoClassDefFoundError was an error you'd receive on all devices older than Android 4.4 when backing out of a video.
- Radio buttons weren't styled correctly on all devices older than Android 5.0 after migrating to AndroidX.
- Position indicators weren't displayed correctly on devices running Android 4.1.
- I enabled TLS v1.1/1.2 for devices stuck on Jelly Bean.

**Comment by blackbox87** [comment]
You're welcome, although I noticed that @friendlyanon thanked me for "fixes for 4.0 and 4.1 devices", but my fixes actually apply to a much wider range of devices. And NewPipe itself doesn't actually support Android 4.0 as the minimum API level is 16.

- InstantiationException was an error you'd receive on all devices when trying to configure PeerTube instances.
- NoClassDefFoundError was an error you'd receive on all devices older than Android 4.4 when backing out of a video.
- Radio buttons weren't styled correctly on all devices older than Android 5.0 after migrating to AndroidX.
- Position indicators weren't displayed correctly on devices running Android 4.1.
- I enabled TLS v1.1/1.2 for devices stuck on Jelly Bean.
